src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -154,13 +58,13 @@ public class TagHelperDescriptor
         /// <remarks>
         /// <c>*</c> at the end of an attribute name acts as a prefix match.
         /// </remarks>
-        public IReadOnlyList<string> RequiredAttributes { get; }
+        public IEnumerable<string> RequiredAttributes { get; [param: NotNull] set; } = Enumerable.Empty<string>();
 
         /// <summary>
         /// HTML element tag names targeted by tag helpers allowed as children.
         /// </summary>
         /// <remarks><c>null</c> indicates all tag helper children are allowed.</remarks>
-        public IReadOnlyList<string> AllowedChildren { get; }
+        public IEnumerable<string> AllowedChildren { get; set; }
```

Comment:
None of these properties match the earlier implementation. Previously all descriptors had their own list instances, created using the `List<T>` copy constructor. Isolation responsibility has shifted here to caller which implies `TagHelperDescriptorResolver` and perhaps `TagHelperDescriptorFactory` need to change more.

(Side note: It's strange to have settable collection properties. But the `IEnumerable<T>` types help a bit. And `AllowedChildren` must have a setter; so this descriptor is consistent.)